src/AirshipBindings.NETStandard/AirshipBindings.NETStandard.Android/Airship.cs

```diff
@@ -112,6 +113,31 @@ public string NamedUser
             }
         }
 
+        private Boolean addedInboxListener = false;
+        private EventHandler onMessageCenterUpdated;
+        public event EventHandler OnMessageCenterUpdated
+        {
+            add
+            {
+                onMessageCenterUpdated += value;
+                if (!addedInboxListener)
```

Comment:
I think you should just add the listener in the constructor and not worry about removing it, or you could check `if (onMessageCenterUpdated == null)` before you add to see if you need to add the listener:

```
add
{
    if (onMessageCenterUpdated == null) {
       MessageCenterClass.Shared().Inbox.AddListener(this);
    }
    onMessageCenterUpdated += value;
}
```

But I think it would be safer to just add it at the beginning once, no reason to add/remove

Reply:
I could put it again at the initialization of Airship, but are you really sure nobody would like to remove the listeners? I'm kind of confused why a listener will be set at the beginning without control and then never removed